src/content_mapblock.cpp

```diff
@@ -172,6 +172,10 @@ static inline void getNeighborConnectingFace(v3s16 p, INodeDefManager *nodedef,
 		*neighbors |= v;
 }
 
+static int NeighbourToIndex(const v3s16 &pos) {
+	return (9 * pos.X + 3 * pos.Y + pos.Z) + 13;
```

Comment:
HyrbidDog. Correct. And those are undocumented preconditions on what x, y, and z can be.